src/write.c

```diff
@@ -46,7 +46,10 @@ avifCodecEncodeOutput * avifCodecEncodeOutputCreate(void)
 {
     avifCodecEncodeOutput * encodeOutput = (avifCodecEncodeOutput *)avifAlloc(sizeof(avifCodecEncodeOutput));
     memset(encodeOutput, 0, sizeof(avifCodecEncodeOutput));
-    avifArrayCreate(&encodeOutput->samples, sizeof(avifEncodeSample), 1);
+    if (avifArrayCreate(&encodeOutput->samples, sizeof(avifEncodeSample), 1) != AVIF_RESULT_OK) {
```

Comment:
The pattern is different from avifEncoderDataCreate() but I do not see why. I suggest both or neither use a goto.

Reply:
This function has only one error path, so it is not necessary to use goto to share error handling code. But I will switch this function to goto for consistency.

Reply:
The reason I use `goto` in this PR is to let error paths share error handling code. This function has only one error path, so it is not necessary to share error handling code. But I will switch this function to `goto` for consistency.